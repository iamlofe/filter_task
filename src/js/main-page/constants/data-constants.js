const LISTRESULTDISM11 = [{
    id: 1,
    title: 'res1',
},
{
    id: 2,
    title: 'res2',
},
{
    id: 3,
    title: 'res3',
},
];
const LISTRESULTDISM12 = [{
    id: 4,
    title: 'res21',
},
{
    id: 5,
    title: 'res22',
},
{
    id: 6,
    title: 'res23',
},
];
const LISTRESULTDISM13 = [{
    id: 7,
    title: 'res31',
},
{
    id: 8,
    title: 'res32',
},
{
    id: 9,
    title: 'res33',
},
];

const LISTRESULTDISM21 = [{
    id: 10,
    title: 'res1',
},
{
    id: 11,
    title: 'res2',
},
{
    id: 12,
    title: 'res3',
},
];
const LISTRESULTDISM22 = [{
    id: 13,
    title: 'res21',
},
{
    id: 14,
    title: 'res22',
},
{
    id: 15,
    title: 'res23',
},
];
const LISTRESULTDISM23 = [{
    id: 16,
    title: 'res31',
},
{
    id: 17,
    title: 'res32',
},
{
    id: 18,
    title: 'res33',
},
];

const LISTRESULTDISM31 = [{
    id: 19,
    title: 'res1',
},
{
    id: 20,
    title: 'res2',
},
{
    id: 21,
    title: 'res3',
},
];
const LISTRESULTDISM32 = [{
    id: 22,
    title: 'res21',
},
{
    id: 23,
    title: 'res22',
},
{
    id: 24,
    title: 'res23',
},
];
const LISTRESULTDISM33 = [{
    id: 25,
    title: 'res31',
},
{
    id: 26,
    title: 'res32',
},
{
    id: 27,
    title: 'res33',
},
];

const DIMENSIONS1 = [{
    id: 28,
    title: 'dim1',
    listResults: LISTRESULTDISM11,
},
{
    id: 29,
    title: 'dim2',
    listResults: LISTRESULTDISM12,
},
{
    id: 30,
    title: 'dim3',
    listResults: LISTRESULTDISM13,
},
];
const DIMENSIONS2 = [{
    id: 31,
    title: 'dim4',
    listResults: LISTRESULTDISM21,
},
{
    id: 32,
    title: 'dim5',
    listResults: LISTRESULTDISM22,
},
{
    id: 33,
    title: 'dim6',
    listResults: LISTRESULTDISM23,
},
];
const DIMENSIONS3 = [{
    id: 34,
    title: 'dim7',
    listResults: LISTRESULTDISM31,
},
{
    id: 35,
    title: 'dim8',
    listResults: LISTRESULTDISM32,
},
{
    id: 36,
    title: 'dim9',
    listResults: LISTRESULTDISM33,
},
];

export const CONTEXTS = [{
    id: 37,
    title: 'context1',
    listsDimensions: DIMENSIONS1,
},
{
    id: 38,
    title: 'context2',
    listsDimensions: DIMENSIONS2,
},
{
    id: 39,
    title: 'context3',
    listsDimensions: DIMENSIONS3,
},
];
