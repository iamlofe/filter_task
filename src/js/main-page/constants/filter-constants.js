export const SearchTypes = {
    BEGIN_WITH: {
        title: 'beginWith',
        symbol: '*'
    },
    EXACT_MATCH: {
        title: 'exactMatch',
        symbol: '**'
    },
    OVERLAP: {
        title: 'overlap',
        symbol: 'A-Z'
    }
};
