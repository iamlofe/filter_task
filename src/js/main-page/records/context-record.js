import { Record } from 'immutable';

const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    demesions: null,
    id: null,

});

export default class CurrentContextRecord extends CurrentContext {
    static parse(info) {
        return new CurrentContextRecord({
            title: info.title,
            id: info.id,
            searchTitle: info.searchTitle,
            searchType: info.searchType
        });
    }
}
