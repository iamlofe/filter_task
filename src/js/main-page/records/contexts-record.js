import { Record, List } from 'immutable';

import RecordDemisions from './dimensions-record';

const Context = new Record({
    id: null,
    title: null,
    listsDimensions: new List(),
    isAccess: true,
    isActive: false,
});

export default class RecordContext extends Context {
    static parse(context) {
        return new RecordContext({
            id: context.id,
            title: context.title,
            listsDimensions:
                new List(context.listsDimensions.map(dimension => RecordDemisions.parse(dimension, context.id))),
        });
    }
}
