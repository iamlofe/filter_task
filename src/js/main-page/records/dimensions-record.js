import {
    Record,
    List,
} from 'immutable';

import RecordResults from './results-record';

const Demisions = new Record({
    id: null,
    title: null,
    listsResults: new List(),
    isActive: false,
    isAccess: false,
    demisionsSave: []
})

export default class RecordDemisions extends Demisions {
    static parse(demisions, contextId) {
        return new RecordDemisions({
            id: demisions.id,
            title: demisions.title,
            contextId: contextId,
            listsResults: new List(demisions.listResults.map(r => RecordResults.parse(r, contextId, demisions.id))),
        })
    }
}