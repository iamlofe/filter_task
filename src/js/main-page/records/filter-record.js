import { Record } from 'immutable';

export const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    isSaving: null,
    isRestoring: null,
    contextIds: null
});
