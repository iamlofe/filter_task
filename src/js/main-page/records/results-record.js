import {
    Record,
    List,
} from 'immutable';

import idGeneration from '../tools/generation-id-tool';

const Results = new Record({
    id: null,
    title: null,
    demisionId: null,
    contextId: null,
    isActive: false,
    isAccess: false,
    resultsSave: []
})


export default class RecordResults extends Results {
    static parse(result, ctxId, dimId) {
        return new RecordResults({
            id: result.id,
            title: result.title,
            demisionId: dimId,
            contextId: ctxId
        })
    }
}