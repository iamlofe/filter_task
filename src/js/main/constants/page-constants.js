import MainPage from 'main-page/components/main-page/main-page';

export default [
    {
        path: '/',
        component: MainPage,
        exact: true
    }
];
