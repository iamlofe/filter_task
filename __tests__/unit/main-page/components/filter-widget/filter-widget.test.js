import React from 'react';
import { shallow, mount } from 'enzyme';
import { List } from 'immutable';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';

import FilterWidget from 'main-page/components/filter-widget/filter-widget';
import { CurrentContext } from 'main-page/records/filter-record';
import { SearchTypes } from 'main-page/constants/filter-constants';
import { getMockStateFilter } from '../../mock-store/mock-filter-reducer';

describe('account component', () => {
    const baseProps = {
        searchString: '',
        filterId: 'test',
        onChangeStateContext: jest.fn(),
        onChangeStateResult: jest.fn(),
        onChooseTypeSearch: jest.fn(),
        onInputTitleSearch: jest.fn(),
        onChangeStateDemission: jest.fn(),
        onToggleWidget: jest.fn(),
        dataFilter: new CurrentContext(),
        filteredDemisions: new List(),
        contexts: new List(),
        filteredResultsWithSort: new List(),
        selectedContext: new List(),
        selectedDemision: new List(),
        selectedResults: new List(),
        filteredList: new List(),
        selectedItems: new List()
    };

    const mockStore = configureMockStore();
    const store = mockStore(getMockStateFilter());

    const filterWidget = (
        <Provider store={store}>
            <MemoryRouter key="test">
                <FilterWidget {...baseProps} />
            </MemoryRouter>
        </Provider>
    );

    it('is shallow rendered without crashing', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        expect(filterSelectedItemsComponent).toMatchSnapshot();
    });

    it('should return truth length of searchTypes', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        expect(filterSelectedItemsComponent.find('.filter-container__sorts').children().length).toEqual(Object.keys(SearchTypes).length);
    });

    it('should handle onSearchStringChange', () => {
        const event = {
            target: {
                value: 'test'
            }
        };

        const filterSelectedItemsComponent = mount(filterWidget);
        filterSelectedItemsComponent.find('.filter-container__field-search').simulate('change', event);
        expect(baseProps.onInputTitleSearch).toBeCalledWith(baseProps.filterId, event.target.value);
    });

    it('should handle onOpenContexts, state isOpenContexts must be true', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        filterSelectedItemsComponent.find('.filter-container__context-drop').simulate('click');
        expect(filterSelectedItemsComponent.state('isOpenContexts')).toEqual(true);
    });

    it('should handle onOpenContexts, state isOpenDemissions must be false', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        filterSelectedItemsComponent.find('.filter-container__context-drop').simulate('click');
        expect(filterSelectedItemsComponent.state('isOpenDemissions')).toEqual(false);
    });

    it('should handle onOpenDemissions , state isOpenContexts must be true', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        filterSelectedItemsComponent.find('.filter-container__demisions-drop').simulate('click');
        expect(filterSelectedItemsComponent.state('isOpenContexts')).toEqual(false);
    });

    it('should handle onOpenDemissions , state isOpenDemissions must be false', () => {
        const filterSelectedItems = <FilterWidget {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        filterSelectedItemsComponent.find('.filter-container__demisions-drop').simulate('click');
        expect(filterSelectedItemsComponent.state('isOpenDemissions')).toEqual(true);
    });
});
