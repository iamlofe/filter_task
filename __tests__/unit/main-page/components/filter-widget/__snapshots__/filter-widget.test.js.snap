// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account component is shallow rendered without crashing 1`] = `
<Draggable
  allowAnyClick={false}
  axis="both"
  bounds={false}
  cancel={null}
  defaultClassName="react-draggable"
  defaultClassNameDragged="react-draggable-dragged"
  defaultClassNameDragging="react-draggable-dragging"
  defaultPosition={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  disabled={false}
  enableUserSelectHack={true}
  grid={
    Array [
      0.01,
      0.01,
    ]
  }
  handle=".filter-container__icon"
  offsetParent={null}
  onDrag={[Function]}
  onMouseDown={[Function]}
  onStart={[Function]}
  onStop={[Function]}
  position={null}
  transform={null}
>
  <div
    className="filter-container"
  >
    <div
      className="filter-container__content"
    >
      <div
        className="filter-container__header"
      >
        <div
          className="filter-container__icon"
        />
        <div
          className="filter-container__label"
        >
          FILTERS
        </div>
        <div
          className="filter-container__close"
          onClick={[MockFunction]}
        />
      </div>
      <div
        className="filter-container__main"
      >
        <div
          className="filter-container__main-content"
        >
          <div
            className="filter-container__context"
          >
            <div
              className="filter-container__context-drop"
              onClick={[Function]}
            >
              <div
                className="filter-container__icon-drop-down"
              />
              <p
                className="filter-container__name"
              >
                Context
              </p>
              <FilterSelectedItems
                filteredList={Immutable.List []}
                selectedItems={Immutable.List []}
              />
            </div>
            <div
              className="filter-container__dropdown-list-context"
            >
              <FilterItems
                displayClass=""
                filterId="test"
                filteredList={Immutable.List []}
                onChangeState={[MockFunction]}
                selectedItems={Immutable.List []}
              />
            </div>
          </div>
          <div
            className="filter-container__demisions"
          >
            <div
              className="filter-container__demisions-drop"
              onClick={[Function]}
            >
              <div
                className="filter-container__icon-drop-down"
              />
              <p
                className="filter-container__name"
              >
                Demisions
              </p>
              <FilterSelectedItems
                filteredList={Immutable.List []}
                selectedItems={Immutable.List []}
              />
            </div>
            <div
              className="filter-container__dropdown-list-demisions"
            >
              <FilterItems
                displayClass=""
                filterId="test"
                filteredList={Immutable.List []}
                onChangeState={[MockFunction]}
                selectedItems={Immutable.List []}
              />
            </div>
          </div>
          <div
            className="filter-container__search"
          >
            <div
              className="filter-container__container-search"
            >
              <div
                className="filter-container__icon-search"
              />
              <div
                className="filter-container__container-field"
              >
                <input
                  className="filter-container__field-search"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <div
                  className="filter-container__sorts"
                >
                  <SearchTypeItem
                    dataFilter={
                      Immutable.Record {
                        "searchTitle": null,
                        "searchType": null,
                        "isSaving": null,
                        "isRestoring": null,
                        "contextIds": null,
                      }
                    }
                    filterId="test"
                    key="beginWith"
                    onChooseTypeSearch={[MockFunction]}
                    symbol="*"
                    type="beginWith"
                  />
                  <SearchTypeItem
                    dataFilter={
                      Immutable.Record {
                        "searchTitle": null,
                        "searchType": null,
                        "isSaving": null,
                        "isRestoring": null,
                        "contextIds": null,
                      }
                    }
                    filterId="test"
                    key="exactMatch"
                    onChooseTypeSearch={[MockFunction]}
                    symbol="**"
                    type="exactMatch"
                  />
                  <SearchTypeItem
                    dataFilter={
                      Immutable.Record {
                        "searchTitle": null,
                        "searchType": null,
                        "isSaving": null,
                        "isRestoring": null,
                        "contextIds": null,
                      }
                    }
                    filterId="test"
                    key="overlap"
                    onChooseTypeSearch={[MockFunction]}
                    symbol="A-Z"
                    type="overlap"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="filter-container__list-results"
          >
            <FilterItems
              displayClass=""
              filterId="test"
              filteredList={Immutable.List []}
              onChangeState={[MockFunction]}
              selectedItems={Immutable.List []}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</Draggable>
`;
